### DISCUSSION TITLE
Purging by UID

### DISCUSSION THREAD

**Discussion by jakubmirejovsky** [strapi-community/strapi-plugin-rest-cache#48] [Q&A] [answered]
Hi, what is the best approach to purge the cache by uid of the cached entity?

This is what we are using now, but I feel it's not the right way:

```
purgeCache: async (rest_route_plural) => {
      const keys = await strapi.plugins["rest-cache"].services.cacheStore.keys()
      const keysToDelete = keys.filter(key => key.includes(rest_route_plural))
      for (let key of keysToDelete) {
        await strapi.plugins["rest-cache"].services.cacheStore.del(key)
      }
    }
```

How to purge all related cache entries to an entity represented by its uid (eg. 'api::item-log.item-log')?

Any help would be appriciated. I tried clearByUid() but with no luck.

**Comment by stafyniaksacha** [comment, marked as answer]
Hello! 

You can use `cacheStore.clearByUid` method
https://github.com/strapi-community/strapi-plugin-rest-cache/blob/0f47f3b9edb934907f12ae9f7e11f7829b986ec0/packages/strapi-plugin-rest-cache/server/services/cacheStore.js#L203
